subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/git/tasks/GitPush.java

```diff
@@ -28,9 +31,16 @@
 
     public void gitPush(GitPushTask task) {
         TokenAvailabilityMessage.logMessage("git push", task.getSecretValue());
-
-        new DefaultProcessRunner(task.getProject().getProjectDir())
+        try {
+            new DefaultProcessRunner(task.getProject().getProjectDir())
                 .setSecretValue(task.getSecretValue())
                 .run(GitPush.gitPushArgs(task.getUrl(), task.getTargets(), task.isDryRun()));
+        } catch (GradleException e) {
+            if (CannotPushToGithubException.matchException(e)) {
```

Comment:
Let's add some tests around that so that we have good coverage for all "interesting" complexity, such as the conditionals. There are a couple of ways, the simplest could be adding a method that handles the exception for a Runnable:

```java
friendlyException(
            new Runnable() {
                public void run() {
                    new DefaultProcessRunner(task.getProject().getProjectDir())
                        .setSecretValue(task.getSecretValue())
                        .run(GitPush.gitPushArgs(task.getUrl(), task.getTargets(), task.isDryRun()));
                }
            });
```

New "friendlyException" method (or however we want to call it :) is easy to unit test.